### PR TITLE
Add PlayerEvent.SleepCheck

### PR DESCRIPTION
**Pull request by shadowfacts**
Event for checking of the player can sleep at a given place.

This allows for things like custom beds and sleeping bag items.

### CODE REVIEW

File: patches/net/minecraft/entity/player/EntityPlayer.java.patch [hunk 1]
```diff
@@ -1,5 +1,14 @@
 --- /net/minecraft/entity/player
 +++ /net/minecraft/entity/player
+@@ -1255,7 +1255,7 @@
+     }
+ 
+     private boolean isInBed() {
+-        return this.worldObj.getBlockState(this.playerLocation).getBlock() == Blocks.BED;
++        return xyz.openmodloader.event.impl.PlayerEvent.SleepCheck.handle(this, playerLocation);
```
Comment (iLexiconn) [inline patches/net/minecraft/entity/player/EntityPlayer.java.patch]: return original code and add `if(true) xyz.openmodloader.event.impl.PlayerEvent.SleepCheck.handle(this, playerLocation);` in front of it

File: src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java [hunk 2]
```diff
@@ -198,4 +200,64 @@ public Stop(EntityPlayer player, Entity tracking) {
...
+     * Fired from {@link EntityPlayer#isInBed()}
+     */
+    public static class SleepCheck extends PlayerEvent {
+
+        /**
+         * The player's position
+         * @see #getPos()
+         */
+        private final BlockPos pos;
+        /**
```
Comment (iLexiconn) [inline src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java]: whiteline lol

File: src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java [hunk 3]
```diff
@@ -198,4 +200,64 @@ public Stop(EntityPlayer player, Entity tracking) {
...
+
+    /**
+     * An even that is fired to check if the player can sleep at the given location.
+     * Fired from {@link EntityPlayer#isInBed()}
+     */
+    public static class SleepCheck extends PlayerEvent {
+
+        /**
+         * The player's position
+         * @see #getPos()
```
Comment (iLexiconn) [inline src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java]: whiteline lol

File: src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java [hunk 4]
```diff
@@ -198,4 +200,64 @@ public Stop(EntityPlayer player, Entity tracking) {
...
+         */
+        private final BlockPos pos;
+        /**
+         * If the player can sleep
+         * By default, this uses the vanilla implementation (if the block at the player's position is a bed, the player can sleep)
+         * @see #getResult()
```
Comment (iLexiconn) [inline src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java]: whiteline lol

File: src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java [hunk 5]
```diff
@@ -198,4 +200,64 @@ public Stop(EntityPlayer player, Entity tracking) {
...
+        /**
+         * @return If the player can sleep
+         */
+        public boolean getResult() {
+            return result;
+        }
+
+        /**
+         * Set's if the player can sleep
+         * @param result {@code true} if the player can sleep, {@code false} if they can't
```
Comment (iLexiconn) [inline src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java]: whiteline lol

File: src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java [hunk 6]
```diff
@@ -198,4 +200,64 @@ public Stop(EntityPlayer player, Entity tracking) {
...
+         */
+        public void setResult(boolean result) {
+            this.result = result;
+        }
+
+        /**
+         * Convenience method for checking if the given player can sleep at the given position
+         * @param player The player
```
Comment (iLexiconn) [inline src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java]: whiteline lol

File: patches/net/minecraft/entity/player/EntityPlayer.java.patch (hunk 1 above)
Comment (shadowfacts) [inline patches/net/minecraft/entity/player/EntityPlayer.java.patch]: Why? It's pointless and only serves the purpose of making the patch smaller.

File: src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java (hunk 2 above)
Comment (phase) [inline src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java]: What on earth is a `whiteline`? [Google](https://www.google.com/search?q=whiteline+programming) returned nothing. Is this an extra new line?

File: patches/net/minecraft/entity/player/EntityPlayer.java.patch (hunk 1 above)
Comment (phase) [inline patches/net/minecraft/entity/player/EntityPlayer.java.patch]: Making a _clear_ patch is better than making a _small_ patch. We don't want patches to be giant, but this is completely fine.

File: src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java (hunk 2 above)
Comment (MrIbby) [inline src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java]: Probably a newline.

Comment (phase) [inline src/main/java/xyz/openmodloader/event/impl/PlayerEvent.java]: This line needs a newline above it, but the rest of the comments are in places that look fine.

**Comment by shadowfacts** [comment]
@iLexiconn Whitespaces fixed

**Pull request merged**